src/metabase/types.cljc

```diff
@@ -253,6 +253,17 @@
 (derive :type/DruidHyperUnique :type/field-values-unsupported)
 (derive :type/DruidJSON :type/field-values-unsupported)
 
+;;; The Snowflake `VARIANT` type is allowed to be anything, so just mark it as deriving from the core root types so
+;;; we're allowed to use any sort of filter with it (whether it makes sense or not). See
+;;; https://docs.snowflake.com/en/sql-reference/data-types-semistructured
+(doseq [t [:type/Number
+           :type/Text
+           :type/Temporal
+           :type/Boolean
+           :type/Collection
+           :type/field-values-unsupported]]
+  (derive :type/SnowflakeVariant t))
```

Comment:
Shouldn't we have some kind of `:type/Any`? If a 3rd party driver wants to have a similar type, are they going to use `:type/SnowflakeVariant` or ask us to introduce it?

Reply:
Yeah I think driver-specific types and the other existing ones actually do belong in driver namespaces, but I didn't want to turn this PR into a refactor PR that started moving unrelated stuff around. I put it here to be with everything else. I'll make a note tho and open a follow-on refactor PR to put stuff in the driver namespaces. I need to make sure it doesn't break anything if you do something like fetch a `:type/SnowflakeVariant` Field before the Snowflake driver namespace gets loaded. We probably need to have the custom type stuff get done at plugin init time which means adding a way to do that from the YAML config files so it happens without loading `metabase.driver.snowflake`